src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -1810,7 +1810,21 @@ public function insert_posts( $items = [] ) {
 							}
 
 							if ( empty( $venue_unique_field ) || ( $lookup_venues_by_title && empty( $venue ) ) ) {
-								$venue = get_page_by_title( $event['Venue']['Venue'], 'OBJECT', Tribe__Events__Venue::POSTTYPE );
+								$venue_query = new WP_Query(
+									[
+										'post_type'      => Tribe__Events__Venue::POSTTYPE,
+										'title'          => $event['Venue']['Venue'],
+										'post_status'    => 'all',
+										'posts_per_page' => 1,
+										'no_found_rows'  => true,
+										'ignore_sticky_posts' => true,
+										'update_post_term_cache' => false,
+										'orderby'        => 'post_date ID',
+										'order'          => 'ASC',
+
+									]
+								);
+								$venue = ! empty( $venue_query->post ) ? $venue_query->post : null;
```

Comment:
Isn't the property we want `posts` ? `post` should be set within the loop and we should not depend being in the loop.

If we are actually ending in the loop because of the updates, this is an issue. I would add a test ensuring that we are not ending up `in_the_loop()`

Reply:
Looking at the source, when `posts` are found, then the first `post` is set at the start. Otherwise, it remains unset (`null`). We're only looking for one post in this case.

Reply:
Seems like you are right !